hackathon_site/event/serializers.py

```diff
@@ -86,6 +86,22 @@ def update(self, instance: Profile, validated_data):
         return serializers.ModelSerializer.update(self, instance, validated_data)
 
 
+class CreateProfileSerializer(ProfileSerializer):
```

Comment:
I may have missed this from our conversation but I don't think you need a new serializer here. In CurrentProfileSerializer you can add your create function in there, no need to make a new serializer.

That also means that you don't need get_serializer_class method in your view and revert back to `serializer_class = CurrentProfileSerializer`